### PR TITLE
DM-40857: Modernize package

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: bin/lsst-build [hunk 1]
```diff
@@ -6,51 +6,55 @@
...
+prepare.add_argument("products", type=str, help="Top-level products to build", nargs="+")
+prepare.add_argument(
+    "--ref", default=[], action="append", type=str, help="An ordered list of refs to check out"
+)
+prepare.add_argument("--repos", type=str, help="YAML file with map of product names to git repository URLs.")
+prepare.add_argument(
+    "--repository-pattern",
+    type=str,
+    default=os.environ.get("REPOSITORY_PATTERN"),
+    help="Python pattern to source git repository " "('|'-delimited; default: $REPOSITORY_PATTERN)",
```
Comment (ktlim) [inline bin/lsst-build]: Can merge these.

File: python/lsst/ci/build.py [hunk 2]
```diff
@@ -60,62 +61,65 @@ class ProgressReporter:
...
         def _build_started(self):
-            self.out.write('%20s: ' % self.product.name)
+            self.out.write(f"{self.product.name:>20s}: ")
             self.out.flush()
             self.progress_bar = self.product.version + " "
             self.t0 = self.t = time.time()
 
         def report_progress(self):
-            # throttled progress reporting
+            # throttled progress reporting.
```
Comment (ktlim) [inline python/lsst/ci/build.py]: Change to docstring rather than comment?

File: python/lsst/ci/build.py [hunk 3]
```diff
@@ -60,62 +61,65 @@ class ProgressReporter:
...
+                self.out.write("(already installed).\n")
                 self.out.flush()
             else:
                 elapsed_time = time.time() - self.t0
                 if retcode:
-                    print("ERROR (%d sec)." % elapsed_time, file=self.out)
-                    print("*** error building product %s." % self.product.name, file=self.out)
-                    print("*** exit code = %d" % retcode, file=self.out)
-                    print("*** log is in %s" % logfile, file=self.out)
+                    print(f"ERROR ({int(elapsed_time)} sec).", file=self.out)
```
Comment (ktlim) [inline python/lsst/ci/build.py]: Kind of weird that this is `int(elapsed_time)` and below on success it's `.1f`.

File: python/lsst/ci/git.py [hunk 4]
```diff
@@ -6,40 +6,43 @@
...
         self.cmd = cmd
         self.output = output
         self.stderr = stderr
 
     def __str__(self):
-        return "Command '%s' returned non-zero exit status %d.\nstdout:\n%s\nstderr:\n%s" % (self.cmd,
-                                                                                             self.returncode,
-                                                                                             self.output,
-                                                                                             self.stderr)
+        return "Command '{}' returned non-zero exit status {}.\nstdout:\n{}\nstderr:\n{}".format(
```
Comment (ktlim) [inline python/lsst/ci/git.py]: Make into an f-string instead?

File: setup.py [hunk 5]
```diff
@@ -4,58 +4,55 @@
...
-        'Development Status :: 3 - Alpha',
-        'Intended Audience :: Developers',
-        'Topic :: Software Development :: Build Tools',
-        'License :: OSI Approved :: MIT License',
-        'Programming Language :: Python :: 3.6',
+        "Development Status :: 3 - Alpha",
+        "Intended Audience :: Developers",
+        "Topic :: Software Development :: Build Tools",
+        "License :: OSI Approved :: MIT License",
+        "Programming Language :: Python :: 3.6",
```
Comment (ktlim) [inline setup.py]: 3.11 now?

File: python/lsst/ci/prepare.py [hunk 6]
```diff
@@ -112,7 +109,7 @@ def from_file(file_object):
             arr = line.split()
             if len(arr) == 4:
                 (name, sha1, version, deps) = arr
-                deps = [dep_name for dep_name in deps.split(',')]
+                deps = list(deps.split(","))
```
Comment (ktlim) [inline python/lsst/ci/prepare.py]: Is the `list()` even needed?

File: python/lsst/ci/prepare.py [hunk 7]
```diff
@@ -147,37 +145,40 @@ class ProductFetcher:
...
             self.repository_patterns = None
         self.no_fetch = no_fetch
         if repos:
             if os.path.exists(repos):
-                with open(repos, 'r', encoding='utf-8') as f:
+                with open(repos, "r", encoding="utf-8") as f:
                     self.repos = yaml.safe_load(f)
             else:
-                raise Exception("YAML repos file '%s' does not exist" % repos)
+                raise Exception(f"YAML repos file {repos!r} does not exist")
```
Comment (ktlim) [inline python/lsst/ci/prepare.py:171]: It's a `str`, so `!r` is not really necessary, right?  (There are some others like this, but they're maybe not as obvious.)

**Review by ktlim — APPROVED** [review]
If we're not actually using markdownlint, can the `.mdl*` files go away?

File: python/lsst/ci/prepare.py (hunk 7 above)
Comment (timj) [reply python/lsst/ci/prepare.py:171]: I was using `%r` to automatically add the single quote for me...

Comment (ktlim) [reply python/lsst/ci/prepare.py:171]: OIC.  Cute.

**Pull request merged**